src/test/java/unit/uk/gov/ida/verifyserviceprovider/services/NonMatchingAssertionServiceTest.java

```diff
@@ -0,0 +1,263 @@
+    }
+
+    @Test
+    public void shouldThrowExceptionIfIssuerValueMissingWhenValidatingIdPAssertion() {
+        Assertion assertion = aMatchingDatasetAssertionWithSignature(emptyList(), anIdpSignature(), "requestId").buildUnencrypted();
+        assertion.setIssuer(anIssuer().withIssuerId(null).build());
+
+        exception.expect(SamlResponseValidationException.class);
+        exception.expectMessage("Assertion with id mds-assertion has missing or blank Issuer.");
+        nonMatchingAssertionService.validateIdPAssertion(assertion, "not-used", IDPSSODescriptor.DEFAULT_ELEMENT_NAME);
+    }
+
+    @Test
+    public void shouldThrowExceptionIfIssuerValueIsBlankWhenValidatingIdPAssertion() {
+        Assertion assertion = aMatchingDatasetAssertionWithSignature(emptyList(), anIdpSignature(), "requestId").buildUnencrypted();
+        assertion.setIssuer(anIssuer().withIssuerId("").build());
+
+        exception.expect(SamlResponseValidationException.class);
+        exception.expectMessage("Assertion with id mds-assertion has missing or blank Issuer.");
+        nonMatchingAssertionService.validateIdPAssertion(assertion, "not-used", IDPSSODescriptor.DEFAULT_ELEMENT_NAME);
+    }
+
+    @Test
+    public void shouldThrowExceptionIfMissingAssertionVersionWhenValidatingIdPAssertion() {
+        Assertion assertion = aMatchingDatasetAssertionWithSignature(emptyList(), anIdpSignature(), "requestId").buildUnencrypted();
+        assertion.setVersion(null);
+
+        exception.expect(SamlResponseValidationException.class);
+        exception.expectMessage("Assertion with id mds-assertion has missing Version.");
+        nonMatchingAssertionService.validateIdPAssertion(assertion, "not-used", IDPSSODescriptor.DEFAULT_ELEMENT_NAME);
+    }
+
+
+    @Test
+    public void shouldThrowExceptionIfAssertionVersionInvalidWhenValidatingIdPAssertion() {
+        Assertion assertion = aMatchingDatasetAssertionWithSignature(emptyList(), anIdpSignature(), "requestId").buildUnencrypted();
+        assertion.setVersion(SAMLVersion.VERSION_10);
+
+        exception.expect(SamlResponseValidationException.class);
+        exception.expectMessage("Assertion with id mds-assertion declared an illegal Version attribute value.");
+        nonMatchingAssertionService.validateIdPAssertion(assertion, "not-used", IDPSSODescriptor.DEFAULT_ELEMENT_NAME);
+    }
+
+    @Test
+    public void shouldNotThrowExceptionsWhenAssertionsAreValid() {
+        List<Assertion> assertions = asList(
+                aMatchingDatasetAssertionWithSignature(emptyList(), anIdpSignature(), "requestId").buildUnencrypted(),
+                anAuthnStatementAssertion(IdaAuthnContext.LEVEL_2_AUTHN_CTX, "requestId").buildUnencrypted());
+
+        nonMatchingAssertionService.validate(assertions,"requestId", LevelOfAssurance.LEVEL_1);
+
+        verify(assertionValidator.getSubjectValidator(), times(2)).validate(any(), any());
```

Comment:
(Same as above comments essentially) - we should `verify` on the `assertionValidator.validate` method rather than this getter